src/wombats/lib/wombat_lib.clj

```diff
@@ -0,0 +1,232 @@
+(def turn-directions [:right :left :about-face])
+(def smoke-directions [:forward :backward :left :right :drop])
+(def game-parameters
+{;; HP Modifiers
+ :collision-hp-damage 10
+ :food-hp-bonus 5
+ :poison-hp-damage 10
+ ;; Score Modifiers
+ :food-score-bonus 10
+ :wombat-hit-bonus 10
+ :zakano-hit-bonus 8
+ :steel-barrier-hit-bonus 2
+ :wood-barrier-hit-bonus 2
+ :wombat-destroyed-bonus 25
+ :zakano-destroyed-bonus 15
+ :wood-barrier-destroyed-bonus 3
+ :steel-barrier-destroyed-bonus 25
+ ;; In game parameters
+ :shot-distance 5})
+
+(defn get-arena-size
+  "Fetchs the size of one side of the arena from the state"
+  [state]
+  (first (:global-dimensions state)))
+
+(defn in?
+  "Return true if coll contains elem"
+  [elem coll]
+  (some #(= elem %) coll))
+
+(defn add-locs
+  "Add local :x and :y coordinates to state matrix"
+  [arena]
+  (reduce
+    #(conj %1
+      (reduce
```

Comment:
Could you maybe use something like: `[map-indexed](https://clojuredocs.org/clojure.core/map-indexed)`?